cmd/cloudFoundryDeploy.go

```diff
@@ -691,7 +691,12 @@ func deployMta(config *cloudFoundryDeployOptions, mtarFilePath string, command c
 		cfDeployParams = append(cfDeployParams, deployParams...)
 	}
 
-	extFileParams, extFiles := handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)
+	var extFileParams []string
+	var extFiles []string
+	if len(config.MtaExtensionDescriptors) > 0 {
+		extFileParams, extFiles = handleMtaExtensionDescriptorsAsList(config.MtaExtensionDescriptors)
+	}
+	extFileParams, extFiles = handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)
```

Comment:
The line 699 always overwrites `extFileParams` and `extFiles` set in line 697. We need an else branch to avoid it
```suggestion
	} else {
		extFileParams, extFiles = handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)
	}
```

Reply:
True, but why did the unit test I added work out anyway?